### PR TITLE
rudimentary support for gash-hopping

### PR DESCRIPTION
**Pull request by horrible-little-slime**
Does not currently handle post-ascension choice adventures, or perming skills. Does not currently encode much information into the Path class. Throws when slightly inconvenienced, which is probably good.

### CODE REVIEW

File: src/ascend.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  print,
+  toInt,
+  use,
+  visitUrl,
+  xpath,
+} from "kolmafia";
+import { $item, $items, $stat, get, have } from ".";
+import { ChateauMantegna } from "./resources";
+
+export enum lifestyle {
```
Comment (gausie) [inline src/ascend.ts:15]: capitalise

File: src/ascend.ts [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  casual = 1,
+  softcore = 2,
+  normal = 2,
+  hardcore = 3,
+}
+
+export class Path {
```
Comment (gausie) [inline src/ascend.ts:22]: put all the path stuff into a Path.ts
combine the list into the class itself as static

File: src/ascend.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+        case $stat`Mysticality`:
+          return 1;
+        case $stat`Moxie`:
+          return 2;
+        default:
+          throw `unknown prime stat for ${playerClass}`;
+      }
+    };
+
+    switch ((moon as string).toLowerCase()) {
```
Comment (gausie) [inline src/ascend.ts:109]: Pull this out into some sort of moon sign function

**Review by gausie — CHANGES_REQUESTED** [review]

File: src/Path.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    true
+  );
+  static oneCrazyRandomSummer: Path = new Path("One Crazy Random Summer", 24);
+  static communityService: Path = new Path("Community Service", 25);
+  static avatarOfWestOfLoathing: Path = new Path(
+    "Avatar of West of Loathing",
+    26,
+    true
+  );
+  static theSource: Path = new Path("The Source", 27);
```
Comment (gausie) [inline src/Path.ts:46]: Title case these and add a `Path.all()` functiont to get all of them for looping and I think we're done

Comment (horrible-little-slime) [reply src/Path.ts]: per our conversation in discord, I am not doing this and am instead making a Paths object again

File: src/Path.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    this.hasCampground = hasCampground;
+    this.hasTerrarium = hasTerrarium;
+    this.stomachSize = stomachSize;
+    this.liverSize = liverSize;
+    this.spleenSize = spleenSize;
+    this.classes = classes;
+  }
+}
+
+export const Paths = {
```
Comment (gausie) [inline src/Path.ts:36]: You might need to `as const` this

**Review by gausie — APPROVED** [review]

File: src/Path.ts (hunk 5 above)
Comment (horrible-little-slime) [reply src/Path.ts:36]: done!

**Pull request merged**